tests/tensorflow2/test_support_dicts.py

```diff
@@ -0,0 +1,42 @@
+# Third Party
+import numpy as np
+import tensorflow as tf
+
+# First Party
+import smdebug.tensorflow as smd
+from smdebug.core.collection import CollectionKeys
+from smdebug.trials import create_trial
+
+
+def get_data():
+    images = np.zeros((64, 224))
+    labels = np.zeros((64, 5))
+    inputs = {"Image_input": images}
+    outputs = {"output-softmax": labels}
+    return inputs, outputs
```

Comment:
can we have some test cases for valid and invalid dicts?
empty dicts,
either inputs/output empty (is this possible?)
are the values always valid? - is there a chance they cant be converted to numpy?

is this applicable only to image inputs?

Reply:
are dicts the only other valid format? can inputs be nested lists or something similar?

Reply:
is this used only with keras fit or can it be used with gradtape too?

Reply:
The flatten call made by smdebug is only after tf runs the same function on the feed_dict.

Both fit and gradtape execute the same call stack

Reply:
> can we have some test cases for valid and invalid dicts?
> empty dicts,
> either inputs/output empty (is this possible?)
> are the values always valid? - is there a chance they cant be converted to numpy?
> 
> is this applicable only to image inputs?

model.fit breaks if the customer passes an empty or invalid feed_dict.